ui/app/mixins/with-component-visibility-detection.js

```diff
@@ -0,0 +1,12 @@
+import Mixin from '@ember/object/mixin';
+
+export default Mixin.create({
+  setupDocumentVisibility: function() {
+    this.set('_visibilityHandler', this.get('visibilityHandler').bind(this));
```

Comment:
Will you ever have > 1 component with this mixin on a screen? Would you have a last one wins situation here?

Reply:
I think all components (or routes) would have their handlers run, right? Which is likely the desired behaviour.

Reply:
Yes to both! Job row and node row both mix this mixin in, so multiple components. Yes all components own their own handler and independently add and remove their own handlers. This prevents accidental side effects when paginating or sorting resulting in one component being destroyed and bringing down all the visibility handlers.